index.js

```diff
@@ -1,58 +1,9 @@
-const registerShortcut = require('hyperterm-register-shortcut');
-const windowSet = new Set([]);
-
-function showWindows (windows, app) {
-  windows.forEach((win, index) => {
-    if (index === windowSet.length-1) {
-      win.focus();
-    } else {
-      win.show();
-    }
-  });
-  if (process.platform !== 'win32') app.show();
-}
+const addWindow = require('./modules/windows').addWindow;
+const setup = require('./modules/setup');
 
-function hideWindows (windows, app) {
-  windows.forEach(win => {
-    if (win.isFullScreen()) return;
-    if (process.platform === 'win32') {
-      win.minimize();
-    } else {
-      win.hide();
-    }
-  });
-  if (process.platform !== 'win32') app.hide(); // Mac OS only (re-focuses the last active app)
-}
-
-function toggleWindowVisibility (app) {
-  const windows = [...windowSet];
-  const focusedWindows = windows.filter(window => window.isFocused());
-  if (focusedWindows.length > 0) {
-    hideWindows(windows, app);
-  } else {
-    showWindows(windows, app);
-  }
-}
-
-function addWindow (window) {
-  windowSet.add(window);
-  window.on('focus', () => {
```

Comment:
I *think* the purpose of this listener was to ensure that the order of the windowSet represented the visual order of the windows, with the most recently focused on top and the least recently focused on the bottom. If you have multiple windows open and you hide them, is the order still correct after you summon with this change?